spec/defines/artifactory_spec.rb

```diff
@@ -3,10 +3,9 @@
 describe 'archive::artifactory' do
   let(:facts) { { os: { family: 'RedHat' }, puppetversion: '4.4.0' } }
 
-  before do
-    MockFunction.new('artifactory_sha1') do |f|
-      f.stub.returns('0d4f4b4b039c10917cfc49f6f6be71e4')
-    end
+  # Mock Puppet V4 API ruby function with a puppet language function equivalent
+  let(:pre_condition) do
+    'function archive::artifactory_checksum($url,$type) { return \'0d4f4b4b039c10917cfc49f6f6be71e4\' }'
```

Comment:
Many thanks to @TraGicCode on slack for suggesting this method of mocking puppet v4 API ruby functions with puppet language functions.